plugin-server/src/main/ingestion-queues/batch-processing/each-batch-ingestion.ts

```diff
@@ -34,9 +34,61 @@ export async function eachBatchIngestion(payload: EachBatchPayload, queue: Inges
         return batches
     }
 
+    if (queue.pluginsServer.updateLatestCapturedAtTeams.size > 0) {
+        await updateLatestEventSentAt(payload, queue)
+    }
+
     await eachBatch(payload, queue, eachMessageIngestion, groupIntoBatchesIngestion, 'ingestion')
 }
 
+async function updateLatestEventSentAt(payload: EachBatchPayload, queue: IngestionConsumer): Promise<void> {
```

Comment:
I think we relax the requirements on this severely and reduce load on postgres that way.

E.g. updating if NULL or once per hour would require a lot less queries. We can then move it into the worker as a new pipeline step as we're fetching the Team object anyways.

This should result in better unit testable code with just as good of a performance.

Reply:
I generally agree here. I love that we are looking out for the user and wanting to convey a new 'last event received at' signal. The only real thing we need right now is `ingested_event` set on team. With that set we have covered our current bases and unblocked ourselves with respect to increasing the time it takes to process events.

As a step two we can maybe do something a bit more clever with webhooks like what @xvello was mentioning with https://docs.datadoghq.com/logs/explorer/live_tail/ and skip Postgres entirely.